### ISSUE TITLE
Can provide a simplification of dgraph api?

### ISSUE THREAD

**Issue by sessionboy** [selmeci/dgraph-tonic#10]
This repo is great, I plan to use it. But I hope it can do more.
The original dgraph api is cumbersome to use, and I always look forward to a simplified version. It may be similar to [prisma](https://github.com/prisma/prisma).

Although dgraph officially launched the easy-to-use GraphQL API, it is not perfect enough, and I look forward to the simple version of GraphQL +-API.

**Comment by selmeci** [comment]
Prisma is really cool project, but I'm not sure that this repository should provide all functions of ORM / ODM....

Prisma client uses [pg](https://www.npmjs.com/package/pg) dependency for communicate with DB, for example. This repository could provide same functionalities as pg in prisma for another ORM/ODM/... project. Maybe with combination of [juniper](https://github.com/graphql-rust/juniper).

What kind of simplification do you want @sessionboy?

**Comment by sessionboy** [comment]
Yeah.I look forward to an API similar to prisma, which can reduce my work, for example: exist、findOne、findMany、findOneAndUpdate、update、upsert、delete etc.

**Comment by lhr0909** [comment]
@sessionboy  We built an ORM that is agnostic to dGraph using TypeScript reflection - https://github.com/xanthous-tech/dgraph-orm . Based on the experience I have with this library, I think right now this is the right level in terms of client, and maybe we could build something in Rust that does similar to our TypeScript implementation in Rust as a separate module. But with my level of Rust I don't know if I can make it work in any reasonable time.
